### ISSUE TITLE
`git` in centos docker image does not support `-C` option

### ISSUE THREAD

**Issue by haiyanmeng** [hmeng-19/runc#5]
`make TEST_DOCKERFILE=script/test_Dockerfile_centos test` failed because the version of git is too old in centos.
The failure is test 26 in bats.

```
not ok 26 spec validator
# (in test file tests/integration/spec.bats, line 77)
#   `[ "$status" -eq 0 ]' failed
```

The centos-based docker image's git version is 1.8.3.1, which does not support `-C` option, and breaks the test 26 in bats.

```
[root@8683d61efe25 runc]# git version
git version 1.8.3.1 
```

**Comment by haiyanmeng** [comment]
@mrunalp ,  this is related to the latest PR which uses the `-C` option of git:

```
run git -C src/runtime-spec reset --hard "${SPEC_COMMIT}"
```

One solution is to stop using `-C` option in `spec.bats`:

```
cd src/runtime-spec
run git reset --hard "${SPEC_COMMIT}"
[ "$status" -eq 0 ]
cd "$HELLO_BUNDLE"
```

Another solution is to install a newer version of `git` in `test_Dockerfile_centos`.

**Comment by haiyanmeng** [comment]
@rhatdan, PTAL.

**Comment by mrunalp** [comment]
Yeah, make sense.

**Comment by rhatdan** [comment]
I think we should do this the old way until RHEL and Centos get an updated git.

**Comment by haiyanmeng** [comment]
@mrunalp , should we fix this when we submit our test_Dockerfile_centos to the upstream? Or just fix it now?

**Comment by rhatdan** [comment]
I would say fix it now, since I believe this will break if you run the test on a centos or RHEL system using the Ubuntu Dockerfile.

**Comment by mrunalp** [comment]
I agree.

**Comment by haiyanmeng** [comment]
@rhatdan , I do not think that currently running the Ubuntu Dockerfile on a cenos or RHEL system would break. 
Because the ubuntu Dockerfile runs `apt-get update` and the `git` in the ubuntu package repo should be new enough to support `-C` option.

**Comment by haiyanmeng** [comment]
@rhatdan, @mrunalp  , the `git -C` problem should not be a problem when a user run the tests of `runc` through docker.

However, if the user wants to run the tests of `runc` ouside of docker and the host os is centos or rhel, the bats test of `runc` should fail because the git version on the host is too old.
